apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -96,6 +97,18 @@ export class UpdateIntegration {
     return result;
   }
 
+  private async setRemoveNovuBranding(command: UpdateIntegrationCommand): Promise<boolean> {
+    const organization = await this.communityOrganizationRepository.findOne({
+      _id: command.organizationId,
+    });
+
+    if (organization && organization.apiServiceLevel !== ApiServiceLevelEnum.FREE) {
+      return command.removeNovuBranding ?? false;
+    }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** I'm wondering whether we should throw an error if the org is on free trier and somebody tries to update the value to `true` 🤔

Reply:
Good point, now this method would return `false` and proceed to set it as `false` in db, ideally we don't want to set this property at all. I will try to implement some check if update is needed.

Reply:
Changed:
- encapsulated the entire related logic in the method
- now the value is updated in DB only if necessary